src/core/geometry/qgswkbtypes.h

```diff
@@ -566,6 +567,147 @@ class CORE_EXPORT QgsWkbTypes
       return Unknown;
     }
 
+    /**
+     * Returns the linear type for a WKB type. For example, for a CompoundCurve, the linear type would be LineString.
+     *
+     * \see curveType()
+     * \see isMultiType()
+     * \see isCurvedType()
+     * \see singleType()
+     * \see flatType()
+     * \see multiType()
+     *
+     * \since QGIS 3.14
+     */
+    static Type linearType( Type type )
```

Comment:
this is definitely a useful method to have. You'll need to add unit tests like this though: https://github.com/qgis/QGIS/blob/297ef133a35c6f41f18a5cecbecc537cb712abfa/tests/src/python/test_qgsgeometry.py#L2751